platform/commonUI/browse/src/windowing/NewTabAction.js

```diff
@@ -30,23 +30,25 @@ define(
         "use strict";
 
         /**
-         * The new window action allows a domain object to be opened
-         * into a new browser window. (Currently this is a stub, present
+         * The new tab action allows a domain object to be opened
+         * into a new browser tab. (Currently this is a stub, present
          * to allow the control to appear in the appropriate location in
          * the user interface.)
          * @constructor
          */
-        function NewWindowAction($window) {
+        function NewTabAction($window) {
             return {
                 /**
-                 * Open the object in a new window (currently a stub)
+                 * Open the object in a new tab
                  */
                 perform: function () {
-                    $window.alert("Not yet functional. This will open objects in a new window.");
+                    // Places the current Url into a variable
+                    // Then the Url is opened using $window
+                    $window.open($window.location.href);
```

Comment:
I would propose handling url generation in a separate service: a urlFor helper can be used in many locations and will give us flexibility down the line if we change url schemes. I'd say an initial implementation could be really dumb, e.g. urlService.urlFor('browse', domainObject).